src/filesystem.c

```diff
@@ -210,6 +212,53 @@ rcutils_mkdir(const char * abs_path)
   return success;
 }
 
+size_t
+rcutils_calculate_directory_size(const char * directory_path, rcutils_allocator_t allocator)
+{
+  size_t dir_size = 0;
+#ifdef _WIN32
+  char * path = rcutils_join_path(directory_path, "*", allocator);
+  WIN32_FIND_DATA data;
+  HANDLE handle = FindFirstFile(path, &data);
+  allocator.deallocate(path, allocator.state);
+  if (handle != INVALID_HANDLE_VALUE) {
+    do {
+      if (strcmp(data.cFileName, ".") != 0 && strcmp(data.cFileName, "..") != 0) {
```

Comment:
Maybe make this a separate function (`directory_entry_is_clean`) or better docs since it's not directly clear what this is doing IMO.

Reply:
I am not convinced that a separate function for this makes sense. I could see a function to check whether a directory is empty, but that wouldn't really apply here as this check really only skips the two file handles (`.` = local folder, `..` = parent folder).

I can update the function with some inline comments if that helps.